mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+# -*- encoding: utf-8 -*-
+'''Display functions'''
+
+import numpy as np
+from scipy.signal import spectrogram
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle, Patch
+from matplotlib.ticker import FuncFormatter, MultipleLocator
+from matplotlib.colors import LinearSegmentedColormap, LogNorm
+
+from .melody import freq_to_voicing
+from .util import midi_to_hz, hz_to_midi
+
+
+def __expand_limits(ax, limits, which='x'):
+    '''Helper function to expand axis limits'''
+
+    if which == 'x':
+        getter, setter = ax.get_xlim, ax.set_xlim
+    elif which == 'y':
+        getter, setter = ax.get_ylim, ax.set_ylim
+    else:
+        raise ValueError('invalid axis: {}'.format(which))
+
+    old_lims = getter()
+    new_lims = list(limits)
+
+    if np.isfinite(old_lims[0]):
+        new_lims[0] = min(old_lims[0], limits[0])
+
+    if np.isfinite(old_lims[1]):
+        new_lims[1] = max(old_lims[1], limits[1])
+
+    setter(new_lims)
+
+
+def segments(intervals, labels, base=None, height=None, text=False,
+             text_kw=None, ax=None, **kwargs):
+    '''Plot a segmentation as a set of disjoint rectangles.
+
+    Parameters
+    ----------
+    intervals : np.ndarray, shape=(n, 2)
+        segment intervals, in the format returned by
+        :func:`mir_eval.io.load_intervals` or
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    labels : list, shape=(n,)
+        reference segment labels, in the format returned by
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    base : number
+        The vertical position of the base of the rectangles.
+        By default, this will be the bottom of the plot.
+
+    height : number
+        The height of the base of the rectangles.
+        By default, this will be the top of the plot (minus `base`).
+
+    text : bool
+        If true, each segment's label is displayed in its
+        upper-left corner
+
+    text_kw : dict
+        If `text==True`, the properties of the text
+        object can be specified here.
+        See `matplotlib.pyplot.text` for valid parameters
```

Comment:
Probably more helpful to point people to `matplotlib.pyplot.Text`, because `Text`'s docstring includes all of the possible kwargs and as far as I can tell `text`'s doesn't.